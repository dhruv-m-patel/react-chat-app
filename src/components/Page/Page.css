@import '../../styles/variables.css';

@keyframes appLogoSpin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

html {
  font-size: 100%;
  height: 100vh;
}

.clickable {
  cursor: pointer;
}

.padTop10px {
  padding-top: 10px;
}

.textRight {
  text-align: right;
}

.darkTheme {
  background-color: var(--backgroundColor);
  color: var(--white);

  & a {
    color: var(--blue);
  }

  & table.table {
    color: var(--white) !important;
  }

  & .text {
    color: var(--white);
  }
}

.lightTheme {
  background-color: var(--white);
  color: var(--black);

  & a {
    color: var(--dark-blue);
  }

  & table.table {
    color: var(--black) !important;
  }

  & .text {
    color: var(--black);
  }
}

.page {
  font-size: var(--baseFontSize);
  font-family: Helvetica, sans-serif;
  min-height: 100vh;

  & a:hover {
    text-decoration: underline;
    cursor: pointer;
  }
}

.app {
  text-align: center;
}

.appLogo {
  height: 40vmin;
  pointer-events: none;
}

.appHeader {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.content {
  text-align: left;
}

@media (prefers-reduced-motion: no-preference) {
  .appLogo {
    animation: appLogoSpin infinite 20s linear;
  }
}
