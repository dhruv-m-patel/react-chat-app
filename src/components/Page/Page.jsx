import React, { useState, useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';
import store from 'store';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faToggleOn } from '@fortawesome/free-solid-svg-icons/faToggleOn';
import { faToggleOff } from '@fortawesome/free-solid-svg-icons/faToggleOff';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import Navbar from 'react-bootstrap/Navbar';
import classnames from 'classnames/bind';
import DefaultHelmet from '../DefaultHelmet';
import styles from './Page.css';

const cx = classnames.bind(styles);

export default function Page({
  title,
  description,
  children,
}) {
  const [hasSwitchedToDarkMode, setHasSwitchedToDarkMode] = useState(undefined);

  const switchToDarkMode = useCallback(() => {
    setHasSwitchedToDarkMode(!hasSwitchedToDarkMode);
    store.set('enableDarkMode', !hasSwitchedToDarkMode);
  }, [hasSwitchedToDarkMode]);

  // Set dark mode initially based on whether user prefers it using os preferences or previously turned it on
  useEffect(() => {
    if (hasSwitchedToDarkMode === undefined) {
      let shouldSetDarkModeInitially = false;
      const darkModeSetting = store.get('enableDarkMode');
      if (darkModeSetting === undefined && typeof window !== 'undefined') {
        shouldSetDarkModeInitially = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
      } else {
        shouldSetDarkModeInitially = darkModeSetting;
      }

      setHasSwitchedToDarkMode(shouldSetDarkModeInitially);
      store.set('enableDarkMode', shouldSetDarkModeInitially);
    }
  }, [hasSwitchedToDarkMode])

  return (
    <Container fluid className={cx('page', { darkTheme: hasSwitchedToDarkMode, lightTheme: !hasSwitchedToDarkMode })}>
      <DefaultHelmet title={title} description={description} />
      <div className={cx('navBar')}>
        <Row>
          <Col xs={{ span: 10 }}>
            <div className={cx('app')}>
              <header className={cx('appHeader')}>
                <img src="/images/react.svg" className={cx('appLogo')} alt="logo" />
                React Chat App
              </header>
            </div>
          </Col>
          <Col xs={{ span: 2 }}>
            <div className={cx('textRight')}>
              Dark Mode
              <FontAwesomeIcon
                icon={hasSwitchedToDarkMode ? faToggleOn : faToggleOff}
                size="2x"
                onClick={switchToDarkMode}
                className={cx('clickable', 'padTop10px')}
              />
            </div>
          </Col>
        </Row>
      </div>
      <br />
      <Container className={cx('content')} fluid>
        {children}
      </Container>
    </Container>
  );
}

Page.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
};

Page.defaultProps = {
  title: undefined,
  description: undefined,
}
