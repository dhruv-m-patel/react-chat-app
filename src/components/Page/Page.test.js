import React from 'react';
import Tester from '../../../tests/Tester';
import Page from './Page';

const tester = new Tester();

describe('Page', () => {
  test('it should render', () => {
    const snapshot = tester.getSnapshot(
      Page,
      { title: 'Test App', description: 'Just testing', children: <div /> },
    );
    expect(snapshot).toMatchSnapshot();
  });
});
