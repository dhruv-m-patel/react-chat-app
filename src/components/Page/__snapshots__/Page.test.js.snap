// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page it should render 1`] = `
<div
  className="Page__page___11SaW Page__lightTheme___ZzJ_L container-fluid"
>
  <div
    className="Page__textRight___80Bv0"
  >
    Dark Mode
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-toggle-off fa-w-18 fa-2x Page__clickable___17a83 Page__padTop10px___2MDcB"
      data-icon="toggle-off"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M384 64H192C85.961 64 0 149.961 0 256s85.961 192 192 192h192c106.039 0 192-85.961 192-192S490.039 64 384 64zM64 256c0-70.741 57.249-128 128-128 70.741 0 128 57.249 128 128 0 70.741-57.249 128-128 128-70.741 0-128-57.249-128-128zm320 128h-48.905c65.217-72.858 65.236-183.12 0-256H384c70.741 0 128 57.249 128 128 0 70.74-57.249 128-128 128z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="Page__app___25fET"
  >
    <header
      className="Page__appHeader___3GNlT"
    >
      <img
        alt="logo"
        className="Page__appLogo___3bWCs"
        src="/images/react.svg"
      />
      <h2>
        React App
      </h2>
      <small>
        A React starter app with SSR support.
      </small>
    </header>
    <br />
    <br />
    <div
      className="Page__content___rCsmc container"
    >
      <div />
    </div>
  </div>
</div>
`;
